python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -659,16 +660,29 @@ def _is_under_min_interval(state, external_sensor):
     return elapsed < external_sensor.min_interval_seconds
 
 
-def _fetch_existing_runs(instance, external_sensor, run_requests):
+def _fetch_existing_runs(
+    instance: DagsterInstance,
+    external_sensor: ExternalSensor,
+    run_requests: Sequence[RunRequest],
+):
     run_keys = [run_request.run_key for run_request in run_requests if run_request.run_key]
 
     if not run_keys:
         return {}
 
+    # fetch runs from the DB with only the run key tag
+    # note: while possible to filter more at DB level with tags - it is avoided here due to observed perf problems
     runs_with_run_keys = instance.get_runs(filters=RunsFilter(tags={RUN_KEY_TAG: run_keys}))
 
+    # filter down to runs with run_key that match the sensor name and its namespace (repository)
     valid_runs = [
-        run for run in runs_with_run_keys if run.tags.get(SENSOR_NAME_TAG) == external_sensor.name
+        run
+        for run in runs_with_run_keys
+        if run.tags.get(SENSOR_NAME_TAG) == external_sensor.name
+        and (
+            external_sensor.get_external_origin().external_repository_origin.get_label()
+            == run.external_pipeline_origin.external_repository_origin.get_label()
```

Comment:
get_id() over get_label() ? I could imagine get_label() truncating in the future or somethign for example, get_id() will always be unique

Reply:
do you mean `selector_id` ? we don't want code loading details which effect the origin id to get in the mix here do we?

Reply:
yeah, get_selector_id, my mistake